src/test/run-make/linker-output-non-utf8/Makefile

```diff
@@ -0,0 +1,12 @@
+-include ../tools.mk
+
+# Make sure we don't ICE if the linker prints a non-UTF-8 error message.
+
+# The zzz it to allow humans to tab complete or glob this thing.
+bad_dir := $(TMPDIR)/zzz$$'\xff'
+
+all:
+	$(RUSTC) library.rs
+	mkdir $(bad_dir)
+	mv $(call STATICLIB,library) $(bad_dir)
```

Comment:
[The reference](https://doc.rust-lang.org/nightly/reference.html#linkage) could use an update, then.  (I'm not going to do it because, as shown here, I don't understand what's going on well enough to write something useful.)

I think at this point I'm just going to disable the test on Windows, if no one objects.  I expect that the linking command won't work there anyway, and it seems like there might be several more cycles of this before I even figure out how to attempt to link a Windows native library.  Someone who can do local testing and actually knows how to develop for Windows can fix it if they want to.